docs/tracingpolicy/README.md

```diff
@@ -0,0 +1,1193 @@
+```yaml
+spec:
+  kprobes:
+  - call: "__x64_sys_read"
+    syscall: true
+    # [...]
+  - call: "__x64_sys_write"
+    syscall: true
+    # [...]
+```
+
+### Arguments
+
+A call contains an `args` field which a list of function arguments to include
+in the trace output. Indeed the BPF code that runs on the hook point requires
+information about the types of arguments of the function being traced to
+properly read, print and filter on its arguments. Currently, this information
+needs to be provided by the user under the `args` section.
+
+Following our example, here is the part that defines the arguments:
+```yaml
+args:
+- index: 0
+  type: "int"
+- index: 1
+  type: "file"
+```
+
+To properly read and hook onto the `fd_install(unsigned int fd, struct file
+*file)` function, the YAML snippet above tells the BPF code that the first
+argument is an `int` and the second argument is a `file`, which is the
+[`struct file`](https://elixir.bootlin.com/linux/latest/source/include/linux/fs.h#L940)
+of the kernel. In this way, the BPF code and its printer can properly collect
+and print the arguments.
+
+These types are sorted by the `index` field, where you can specify the order.
+The indexing starts with 0. So, `index: 0` means, this is going to be the first
+argument of the function, `index: 1` means this is going to be the second
+argument of the function, etc.
+
+Note that for some args types, `char_buf` and `char_iovec`, there is an
+additional field named `returnCopy` and a `sizeArgIndex` field to specify to
+use one of the arguments to read the buffer correctly from the BPF code.
+
+This is the length that the BPF program is looking for in order to read the
+`char_buf`. So, `size_t` is the number of `char_buf` elements stored in the
+pointer of the second argument. Similarly, if we would like to capture the the
+`__x64_sys_writev(long, iovec *, vlen)` syscall, then `iovec` has a size of
+`vlen`, which is going to be the 3rd argument.
+
+Please note that `sizeArgIndex` is inconsistent at the moment and does not take
+the index, but the number of the index (or index + 1). So if the size is the
+third argument, index 2, the value should be 3.
+
+```yaml
+- call: "__x64_sys_write"
+  syscall: true
+  args:
+  - index: 0
+    type: "int"
+  - index: 1
+    type: "char_buf"
+    returnCopy: true
+    sizeArgIndex: 3
+  - index: 2
+    type: "size_t"
+```
+
+Note, that you can specify which arguments you would like to print from a
+specific syscall. For example if you don't care about the file descriptor,
+which is the first `int` argument with `index: 0` and just want the `char_buf`,
+what is written, then you can leave this section out and just define:
+```yaml
+args:
+- index: 1
+  type: "char_buf"
+  returnCopy: true
+  sizeArgIndex: 3
+- index: 2
+  type: "size_t"
+```
+This tells the printer to skip printing the `int` arg because it's not useful.
```

Comment:
Can we include which arguments are supported atm? Or just put a pointer to the CRD?

Reply:
Yes I wanted to add details on the various args we support!

Reply:
I just added a link to the CRD but we could do better!